### PR TITLE
migrate Lists from RxJava to Kotlin coroutines

### PR DESCRIPTION
**Pull request by connyduck**
slowly but surely getting rid of Rx

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/viewmodel/ListsViewModel.kt [hunk 1]
```diff
@@ -39,86 +42,94 @@ internal class ListsViewModel @Inject constructor(private val api: MastodonApi)
...
     data class State(val lists: List<MastoList>, val loadingState: LoadingState)
 
-    val state: Observable<State> get() = _state
-    val events: Observable<Event> get() = _events
-    private val _state = BehaviorSubject.createDefault(State(listOf(), LoadingState.INITIAL))
-    private val _events = PublishSubject.create<Event>()
+    val state: Flow<State> get() = _state
+    val events: Flow<Event> get() = _events
+    private val _state = MutableStateFlow(State(listOf(), LoadingState.INITIAL))
+    private val _events = MutableSharedFlow<Event>(replay = 0, extraBufferCapacity = 1, onBufferOverflow = BufferOverflow.DROP_LATEST)
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/viewmodel/ListsViewModel.kt:48]: Shouldn't it be `DROP_OLDEST`?
Also
>Values in state flow are conflated using [Any.equals](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/equals.html) comparison in a similar way to [distinctUntilChanged](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/distinct-until-changed.html) operator. It is used to conflate incoming updates to [value](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-mutable-state-flow/value.html) in [MutableStateFlow](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-mutable-state-flow/index.html) and to suppress emission of the values to collectors when new value is equal to the previously emitted one

which is not what we want? We want to receive each event I think?

Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/viewmodel/ListsViewModel.kt:48]: Oh sorry it's `StateFlow` that compares

Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/viewmodel/ListsViewModel.kt:48]: which it is :facepalm:

Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/viewmodel/ListsViewModel.kt:48]: No it's `MutableStateFlow` :facepalm: :facepalm:

Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/viewmodel/ListsViewModel.kt]: now it should be ok

**Review by charlag — APPROVED** [review]
Yeah looks good!

**Pull request merged**